lib/IMAP/MessageMapper.php

```diff
@@ -241,6 +241,30 @@ public function addFlag(Horde_Imap_Client_Socket $client,
 		);
 	}
 
+	public function getRawMessage(Horde_Imap_Client_Socket $client,
+								string $mailbox,
+								int $id): ?string {
+		$query = new Horde_Imap_Client_Fetch_Query();
+		$query->uid();
+		$query->fullText();
+
+
+		$result = iterator_to_array($client->fetch($mailbox, $query, [
+			'ids' => new Horde_Imap_Client_Ids($id),
+		]), false);
+
+		$msg = array_map(function (Horde_Imap_Client_Data_Fetch $result) use ($client, $mailbox) {
+			return new IMAPMessage(
+				$client,
+				$mailbox,
+				$result->getUid(),
+				$result
+			);
+		}, $result);
+
+		return $msg[0]->getRawMessage();
```

Comment:
check for empty `$msg` otherwise this will do :boom: when the UID doesn't exist :)